drivers/dai/intel/ssp/ssp.c

```diff
@@ -1533,6 +1533,92 @@ static int dai_ssp_set_config_tplg(struct dai_intel_ssp *dp, const struct dai_co
 	return ret;
 }
 
+static int dai_ssp_parse_aux_data(struct dai_intel_ssp *dp, const void *spec_config)
+{
+	const struct dai_intel_ipc4_ssp_configuration_blob_ver_1_5 *blob = spec_config;
+	int aux_tlv_size = sizeof(struct ssp_intel_aux_tlv);
+	int hop, i, j, cfg_len, pre_aux_len, aux_len;
+	struct ssp_intel_aux_tlv *aux_tlv;
+	struct ssp_intel_mn_ctl *mn;
+	struct ssp_intel_clk_ctl *clk;
+	struct ssp_intel_tr_ctl *tr;
+	struct ssp_intel_run_ctl *run;
+	struct ssp_intel_node_ctl *node;
+	struct ssp_intel_sync_ctl *sync;
+	struct ssp_intel_ext_ctl *ext;
+#ifdef CONFIG_SOC_SERIES_INTEL_ACE
+	struct ssp_intel_link_ctl *link;
+#endif
+	uint8_t *aux_ptr;
+
+	cfg_len = blob->size;
+	pre_aux_len = sizeof(*blob) + blob->i2s_mclk_control.mdivrcnt * sizeof(uint32_t);
+	aux_len = cfg_len - pre_aux_len;
+	aux_ptr = (uint8_t *)blob + pre_aux_len;
+
+	if (aux_len <= 0)
+		return 0;
+
+	for (i = 0; i < aux_len; i += hop) {
+		aux_tlv = (struct ssp_intel_aux_tlv *)(aux_ptr);
+		switch (aux_tlv->type) {
+		case SSP_MN_DIVIDER_CONTROLS:
+			mn = (struct ssp_intel_mn_ctl *)&aux_tlv->val;
```

Comment:
Sorry, late comment. Should we have length validation for invalid blobs where sizeof(struct ssp_intel_mn_ctl) goes outside aux_len? We'd end up accessing beyond the blob in thsi case.

Reply:
ok I'll add a function to check for the struct sizes during parsing.

Reply:
@kv2019i added a check function to check for some cases of malformed structs in parsing.